lib/postgrex_wal/gen_stage/pg_source.ex

```diff
@@ -0,0 +1,189 @@
+defmodule PostgrexWal.GenStage.PgSource do
+  @moduledoc false
+
+  use Postgrex.ReplicationConnection, restart: :permanent, shutdown: 10_000
+  use TypedStruct
+  require Logger
+
+  @typep step() :: :disconnected | :streaming
+  typedstruct enforce: true do
+    field :publication_name, String.t()
+    field :slot_name, String.t()
+    field :final_lsn, integer(), default: 0
+    field :step, step(), default: :disconnected
+  end
+
+  @doc ~S"""
+  ## Sample (name field is optional)
+
+   opts = [
+     name: :my_pg_source,
+     publication_name: "mypub1",
+     slot_name: "myslot1",
+     host: "localhost",
+     database: "r704_development",
+     username: "jswk"
+   ]
+  """
+
+  @typep opts() :: [
+           {:name, GenServer.name()},
+           {:publication_name, String.t()},
+           {:slot_name, String.t()},
+           {:host, String.t()},
+           {:database, String.t()},
+           {:username, String.t()}
+         ]
+  @spec start_link(opts()) :: {:ok, pid()} | {:error, Postgrex.Error.t() | term()}
+  def start_link(opts) do
+    # Automatically reconnect if we lose connection.
+    {name_opts, opts} = Keyword.split(opts, [:name])
+    extra_opts = [auto_reconnect: true]
+
+    {publication_name, opts} = Keyword.pop!(opts, :publication_name)
+    {slot_name, opts} = Keyword.pop!(opts, :slot_name)
+
+    Postgrex.ReplicationConnection.start_link(
+      __MODULE__,
+      struct(__MODULE__, publication_name: publication_name, slot_name: slot_name),
+      extra_opts ++ name_opts ++ opts
+    )
+  end
+
+  # api functions()
+
+  @spec stop(GenServer.server()) :: :ok
+  def stop(server) do
+    GenServer.stop(server)
+  end
+
+  @spec async_ack(GenServer.server(), integer) :: {:ack, integer}
```

Comment:
要不类型都加一个 `()` 吧，统一一下风格
```suggestion
  @spec async_ack(GenServer.server(), integer()) :: {:ack, integer()}
```